.gitignore

```diff
@@ -1,8 +1,12 @@
+@@ -1,11 +1,9 @@
 *.o
 *.so
+.gitignore
```

Comment:
I don't think we should put `.gitignore` there.

Reply:
I know it seems like redundant but I added .gitignore to maintain consistency and prevent accidental commits of this file, ensuring clarity in our version control process. And actually it also serves as a reminder for all contributors that we should ignore this file itself to prevent accidental commits.

Reply:
@rafsun42 +1

Reply:
I don't know what's wrong with my changes, it gives clarity and better understanding of the file.